internal/config/mount_config.go

```diff
@@ -130,6 +143,13 @@ type MountConfig struct {
 	AuthConfig          `yaml:"auth-config"`
 	EnableHNS           `yaml:"enable-hns"`
 	FileSystemConfig    `yaml:"file-system"`
+
+	// MetadataPrefetchMode indicates whether or not to prefetch the metadata of the mounted bucket at the time of mounting the bucket.
+	// Supported values: MetadataPrefetchModeDisabled, MetadataPrefetchModeSynchronous, and MetadataPrefetchModeAsynchronous.
+	// Any other values will return error on mounting.
+	// This is applicable only to single-bucket mount-points, and not to dynamic-mount points. This is because dynamic-mounts don't mount the bucket(s) at the time of
+	// gcsfuse command itself, which flag is targeted at.
+	MetadataPrefetchMode metadataPrefetchModeValue `yaml:"metadata-prefetch-mode"`
```

Comment:
Should this field be a member of MetadataCacheConfig type since this is about prepopulating the metadata-cache?

Reply:
I am not in favour of this change, as this field does not configure metadata-cache, nor is its effect limited to only metadata-cache (it also creates all the inodes)

Let's discuss this offline.

Reply:
This is now a moot point as there is no such config-file parameter, and it's been replaced now with a cli flag, which don't have nesting.